*** target/xdoclet/META-INF-hsql/jbosscmp-jdbc.xml	Thu Feb 26 14:28:58 2004
--- target/xdoclet/META-INF-db2/jbosscmp-jdbc.xml	Thu Feb 26 14:54:26 2004
***************
*** 4,10 ****
  <jbosscmp-jdbc>
     <defaults>
       <datasource>java:/DefaultDS</datasource>
!      <datasource-mapping>Hypersonic SQL</datasource-mapping>
       <create-table>false</create-table>
     </defaults>
  
--- 4,10 ----
  <jbosscmp-jdbc>
     <defaults>
       <datasource>java:/DefaultDS</datasource>
!      <datasource-mapping>DB2</datasource-mapping>
       <create-table>false</create-table>
     </defaults>
  
***************
*** 77,83 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 77,84 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for mwl_item_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 129,135 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 130,137 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for instance_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 179,185 ****
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 181,188 ----
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="pk-sql">
! 		<attribute name="pk-sql">values nextval for ae_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 226,232 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 229,236 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for patient_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 293,299 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 297,304 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for study_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 345,351 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 350,357 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for series_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 428,434 ****
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 434,441 ----
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for move_order_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 488,494 ****
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 495,502 ----
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for files_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 525,531 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 533,540 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for code_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
