*** target/xdoclet/META-INF-hsql/jbosscmp-jdbc.xml	Sun Jan 18 00:50:47 2004
--- target/xdoclet/META-INF-db2/jbosscmp-jdbc.xml	Sun Jan 18 00:57:26 2004
***************
*** 4,10 ****
  <jbosscmp-jdbc>
     <defaults>
       <datasource>java:/DefaultDS</datasource>
!      <datasource-mapping>Hypersonic SQL</datasource-mapping>
       <create-table>false</create-table>
     </defaults>
  
--- 4,10 ----
  <jbosscmp-jdbc>
     <defaults>
       <datasource>java:/DefaultDS</datasource>
!      <datasource-mapping>DB2</datasource-mapping>
       <create-table>false</create-table>
     </defaults>
  
***************
*** 62,68 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 62,69 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for instance_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 124,130 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 125,132 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for study_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 171,177 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 173,180 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for patient_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 254,260 ****
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 257,264 ----
  			</read-ahead>
  	     </query>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for move_order_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 335,341 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 339,346 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for series_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 382,388 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 387,394 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for file_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
***************
*** 419,425 ****
  
          </cmp-field>
  
! 	<entity-command name="hsqldb-fetch-key">
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
--- 425,432 ----
  
          </cmp-field>
  
! 	<entity-command name="pk-sql">
!          <attribute name="pk-sql">values nextval for code_pk_seq</attribute>
  	</entity-command>
  <!-- jboss 3.2 features -->
  <!-- optimistic locking does not express the exclusions needed -->
