*** jboss-3.2.2/server/pacs/conf/standardjbosscmp-jdbc.xml	Sat Nov 22 20:20:19 2003
--- standardjbosscmp-jdbc.xml	Tue Nov 25 23:37:08 2003
***************
*** 503,508 ****
--- 503,513 ----
              <sql-type>VARCHAR(254)</sql-type>
           </mapping>
           <mapping>
+             <java-type>java.util.Date</java-type>
+             <jdbc-type>TIMESTAMP</jdbc-type>
+             <sql-type>TIMESTAMP(</sql-type>
+          </mapping>
+          <mapping>
              <java-type>java.sql.Date</java-type>
              <jdbc-type>DATE</jdbc-type>
              <sql-type>DATE</sql-type>
***************
*** 519,525 ****
           </mapping>
           <mapping>
              <java-type>java.lang.Object</java-type>
!             <jdbc-type>JAVA_OBJECT</jdbc-type>
              <sql-type>BLOB(2000)</sql-type>
           </mapping>
        </type-mapping>
--- 524,530 ----
           </mapping>
           <mapping>
              <java-type>java.lang.Object</java-type>
!             <jdbc-type>VARBINARY</jdbc-type>
              <sql-type>BLOB(2000)</sql-type>
           </mapping>
        </type-mapping>
